Adding SDK code for AzureAnalysisServices

athipp opened the pull request:
This project is for enabling client code generation for Azure Analysis
services. The swagger spec for that is here
https://github.com/Azure/azure-rest-api-specs/pull/579#issuecomment-251185810

https://github.com/Azure/azure-rest-api-specs/blob/master/arm-asazure/2016-05-06/swagger/asazure.json

This checkin contains
- Generatedcode for the swagger spec
- InmemoryTests for the APIs.
  Note: For now we have sufficient coverage with inmemroy tests. I would
  like to go ahead with powershell development. So scenario tests is
  coming in further checkin.

athipp committed:
Adding SDK code for AzureAnalysisServices

This project is for enabling client code generation for Azure Analysis
services. The swagger spec for that is here

https://github.com/Azure/azure-rest-api-specs/blob/master/arm-asazure/2016-05-06/swagger/asazure.json

This checkin contains
- Generatedcode for the swagger spec
- InmemoryTests for the APIs.
Note: For now we have sufficient coverage with inmemroy tests. I would
like to go ahead with powershell development. So scenario tests is
coming in further checkin.

azuresdkci commented:
Can one of the admins verify this patch?

azurecla commented:
Hi __@athipp__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (athipp). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

markcowl commented:
@azuresdkci add to whitelist

shahabhijeet commented on src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/InMemoryTests/ServersTests.InMemory.cs:
```diff
@@ -0,0 +1,526 @@
+            acceptedResponse.Headers.Add("x-ms-request-id", "1");
+            acceptedResponse.Headers.Add("Location", @"http://someLocationURL");
+
+            var okResponse = new HttpResponseMessage(HttpStatusCode.OK)
```
>HttpResponseMessage [](start = 33, length = 19)

Any reason you are not using HttpRecorder to record your tests?

athipp replied:
I am working on it and will submit a separate PR for that. SPN based authentication doesn;t work so I need to figure out how to do username based auth. I want to make progress on powershell code though. thats why I submitted this.

shahabhijeet commented on src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/Helpers/AnalysisServicesTestUtilities.cs:
```diff
@@ -0,0 +1,215 @@
+            if (IsTestTenant)
+            {
+                analysisServicesClient = new AnalysisServicesClient(new TokenCredentials("xyz"), GetHandler());
+                analysisServicesClient.SubscriptionId = testSubscription;
```
>testSubscription [](start = 56, length = 16)

Any reason you are not using TestEnvironment and connection string to set all your credentials and subscription information?

athipp replied:
I am working on it and will submit a separate PR for that. SPN based authentication doesn;t work so I need to figure out how to do username based auth. I want to make progress on powershell code though. thats why I submitted this first

shahabhijeet commented:
@azuresdkci retest this please

athipp committed:
Updated SDK with names

Updated SDK with some name changes to reflect the recent swagger spec
changes.

athipp committed:
Adding httprecorder sessions and more tests

- Added scenario tests with httprecorder session logs.
- Updated SDK with changes reflecting swagger spec.
- bug fixes.

athipp committed:
Adding resume suspend apis

athipp committed:
Merge remote-tracking branch 'refs/remotes/Azure/AutoRest' into AutoRest

shahabhijeet commented on src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/Helpers/AnalysisServicesTestUtilities.cs:
```diff
@@ -0,0 +1,200 @@
+            //if (!IsTestTenant)
+            //{
+            //    resourcesClient.ResourceGroups.Delete(resourceGroupName);
+            //}
```
Remove commented code

shahabhijeet commented on src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/Properties/AssemblyInfo.cs:
```diff
@@ -0,0 +1,36 @@
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
```
remove commented code

athipp replied:
done

shahabhijeet commented:
> using Microsoft.Azure.Management.DataLake.Store;

I think this might be a mistake, can you revert this file and not combine with your PR

---
Refers to: src/ResourceManagement/DataLake.Store/DataLakeStore.Tests/ScenarioTests/AccountOperationTests.cs:16 in 8b97eea. [](commit_id = 8b97eeaa35da64c32ab317a8b8bb0a2ad7667d23, deletion_comment = False)

shahabhijeet commented on src/ResourceManagement/AnalysisServices/AnalysisServices.Tests/ScenarioTests/ServerOperationsTests.cs:
```diff
@@ -0,0 +1,163 @@
+    public class ServerOperationsTests : TestBase
+    {
+        [Fact]
+        public void CreateGetUpdateDeleteTest()
```
>CreateGetUpdateDeleteTest [](start = 20, length = 25)

Can this be split into multiple smaller tests? If possible.
For e.g. can the update/delete be separate tests reusing the create from a common code.
this is just a guidance, because in case if the test failed, you won't know if the create failed or update or deleting the account failed.

athipp replied:
For this release I would like to go with one test that test the happy path.

athipp commented:
Closing this PR for now. Will open it once comments are fixed.

athipp committed:
Addressing review comments. Deleting unsued code.

azurecla commented:
Hi __@athipp__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (athipp). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

athipp committed:
Merge remote-tracking branch 'refs/remotes/Azure/AutoRest' into AutoRest

azurecla commented:
Hi __@athipp__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (athipp). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

athipp committed:
Regenerated for Suspend and Resume long running operations
